sphinx/ext/intersphinx.py

```diff
@@ -29,50 +30,90 @@
 from urllib.parse import urlsplit, urlunsplit
 
 from docutils import nodes
-from docutils.utils import relative_path
 
 import sphinx
 from sphinx.addnodes import pending_xref
 from sphinx.builders.html import INVENTORY_FILENAME
 from sphinx.deprecation import _deprecation_warning
 from sphinx.errors import ExtensionError
-from sphinx.locale import _, __
+from sphinx.locale import __
 from sphinx.transforms.post_transforms import ReferencesResolver
 from sphinx.util import logging, requests
 from sphinx.util.docutils import CustomReSTDispatcher, SphinxRole
-from sphinx.util.inventory import InventoryFile
+from sphinx.util.inventory import InventoryFile, InventoryItemSet
 
 if TYPE_CHECKING:
     from collections.abc import Iterable
     from types import ModuleType
     from typing import IO, Any, Union
 
-    from docutils.nodes import Node, TextElement, system_message
+    from docutils.nodes import Element, Node, TextElement, system_message
     from docutils.utils import Reporter
 
     from sphinx.application import Sphinx
     from sphinx.config import Config
     from sphinx.domains import Domain
     from sphinx.environment import BuildEnvironment
-    from sphinx.util.typing import ExtensionMetadata, Inventory, InventoryItem, RoleFunction
+    from sphinx.util.typing import ExtensionMetadata, Inventory, RoleFunction
 
     InventoryCacheEntry = tuple[Union[str, None], int, Inventory]
 
 logger = logging.getLogger(__name__)
 
 
-class InventoryAdapter:
-    """Inventory adapter for environment"""
+def process_disabled_reftypes(env: BuildEnvironment) -> None:
+    # Compile intersphinx_disabled_reftypes into datastructures
+    # easier to check during reference resolution.
+    # It is a separate function so the tests can use it.
+    env.intersphinx_all_disabled = False  # type: ignore[attr-defined]
+    env.intersphinx_all_domain_disabled = set()  # type: ignore[attr-defined]
+    env.intersphinx_disabled_per_domain = {}  # type: ignore[attr-defined]
+    for d in env.config.intersphinx_disabled_reftypes:
+        if d == '*':
+            env.intersphinx_all_disabled = True  # type: ignore[attr-defined]
+        elif ':' in d:
+            domain, typ = d.split(':', 1)
+            if typ == '*':
+                env.intersphinx_all_domain_disabled.add(domain)  # type: ignore[attr-defined]
+            else:
+                env.intersphinx_disabled_per_domain.setdefault(  # type: ignore[attr-defined]
+                    domain, []).append(typ)
+
+
+class EnvAdapter:
```

Comment:
Could we not change the public name? nor the interface that people might be using in their extensions? 

For 8.x, we would add a deprecation warning for InventoryAdapter and use the class you are suggesting instead. But for compat, I would like not to change any "public" names.

Reply:
Yes, I have reinstated the old class and added a deprecation warning.